frappe/core/doctype/communication/communication.py

```diff
@@ -494,7 +494,11 @@ def get_permission_query_conditions_for_communication(user):
 		return None
 	else:
 		accounts = frappe.get_all(
-			"User Email", filters={"parent": user}, fields=["email_account"], distinct=True, order_by="idx"
+			"User Email",
+			filters={"parent": user},
+			fields=["email_account", "idx"],
```

Comment:
This isn't the same. Added idx in field kinda makes `distinct=True` ineffective.

We don't care about idx here, we want distinct email accounts with sorted order of `idx`.